rules/terraformrules/terraform_naming_convention.go

```diff
@@ -0,0 +1,300 @@
+				}
+
+				runner.EmitIssue(
+					r,
+					fmt.Sprintf(message, local.Name, localsNameValidator.Format),
+					local.DeclRange,
+				)
+			}
+		}
+	}
+
+	if moduleNameValidator != nil {
+		for _, module := range runner.TFConfig.Module.ModuleCalls {
+			if !moduleNameValidator.Regexp.MatchString(module.Name) {
+				var message string
+				if moduleNameValidator.IsNamedFormat {
+					message = "module name `%s` must match the following format: %s"
+				} else {
+					message = "module name `%s` must match the following RegExp: %s"
+				}
+
+				runner.EmitIssue(
+					r,
+					fmt.Sprintf(message, module.Name, moduleNameValidator.Format),
+					module.DeclRange,
+				)
+			}
+		}
+	}
+
+	if outputNameValidator != nil {
+		for _, output := range runner.TFConfig.Module.Outputs {
+			if !outputNameValidator.Regexp.MatchString(output.Name) {
+				var message string
+				if outputNameValidator.IsNamedFormat {
+					message = "output name `%s` must match the following format: %s"
+				} else {
+					message = "output name `%s` must match the following RegExp: %s"
+				}
+
+				runner.EmitIssue(
+					r,
+					fmt.Sprintf(message, output.Name, outputNameValidator.Format),
+					output.DeclRange,
+				)
+			}
+		}
+	}
+
+	if resourceNameValidator != nil {
+		for _, resource := range runner.TFConfig.Module.ManagedResources {
+			if !resourceNameValidator.Regexp.MatchString(resource.Name) {
+				var message string
+				if resourceNameValidator.IsNamedFormat {
+					message = "resource name `%s` must match the following format: %s"
+				} else {
+					message = "resource name `%s` must match the following RegExp: %s"
+				}
+
+				runner.EmitIssue(
+					r,
+					fmt.Sprintf(message, resource.Name, resourceNameValidator.Format),
+					resource.DeclRange,
+				)
+			}
+		}
+	}
+
+	if variableNameValidator != nil {
+		for _, variable := range runner.TFConfig.Module.Variables {
+			if !variableNameValidator.Regexp.MatchString(variable.Name) {
+				var message string
+				if variableNameValidator.IsNamedFormat {
+					message = "variable name `%s` must match the following format: %s"
+				} else {
+					message = "variable name `%s` must match the following RegExp: %s"
+				}
+
+				runner.EmitIssue(
+					r,
+					fmt.Sprintf(message, variable.Name, variableNameValidator.Format),
+					variable.DeclRange,
+				)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (config *BlockFormatConfig) getNameValidator() (*NameValidator, error) {
+	return getNameValidator(config.Custom, config.Format)
+}
+
+func (config *terraformNamingConventionRuleConfig) getNameValidator() (*NameValidator, error) {
+	return getNameValidator(config.Custom, config.Format)
+}
+
+func getNameValidator(custom string, format string) (*NameValidator, error) {
+	// Prefer custom format if specified
+	if custom != "" {
+		nameValidator := &NameValidator{
+			IsNamedFormat: false,
+			Format:        custom,
+			Regexp:        regexp.MustCompile(custom),
```

Comment:
`MustCompile` panics if the `custom` pattern cannot be parsed.
https://golang.org/pkg/regexp/#MustCompile